cli/postgres/README.md

```diff
@@ -84,16 +74,80 @@ The tool:
 4. Renders the definition as GraphQL SDL
 5. Outputs the SDL to stdout or a file
 
-## Advanced Usage
+## Configuration
 
-### Using a Custom Extension URL
+Configure the introspection command using a TOML configuration file. Include these essential settings:
 
-If you need to use a custom extension URL instead of the official version:
+```toml
+# The URL of the extension, which appears at the top of the GraphQL SDL.
+extension_url = "https://grafbase.com/extensions/postgres/0.3.0"
 
-```bash
-grafbase-postgres --database-url "postgres://user:pass@localhost:5432/mydb" introspect --extension-url "https://example.com/my-postgres-extension"
+# The default schema, which we'll omit from the SDL output.
+# Defaults to "public" if you don't specify it
+default_schema = "public"
+
+# The name of the database the virtual subgraph should use. This
+# maps to a Postgres database name in your gateway configuration.
+# Defaults to "default" if you don't specify it
+database_name = "default"
+```
+
+### Exposing Views
+
+PostgreSQL views require additional configuration because the information schema doesn't provide details about unique constraints, nullability, or relations. To make a view visible in your GraphQL SDL, you must define at least one unique key.
+
+#### Unique Key Definitions
+
+```toml
+[schemas.public.views.my_view]
+# The order of columns matters - match the order in the underlying query/table.
+# Define compound keys like this:
+unique_keys = [["user_name", "user_id"]]
+
+# structure: schemas.<schema_name>.views.<view_name>.columns.<column_name>
+[schemas.public.views.my_view.columns.user_name]
+# Defaults to unique if you omit this setting
+nullable = false
+# Define a single-column unique key here
+unique = false
+# Customize the GraphQL field name:
+rename = "name_user"
+# Add a description that appears as a comment in the GraphQL schema:
+description = "The name of the user"
+
+[schemas.public.views.my_view.columns.user_id]
+nullable = false
 ```
 
+The introspection will fail if you reference any non-existent schemas, views, or columns.
+
+#### Relation Definitions
+
+```toml
+# structure: schemas.<schema_name>.views.<view_name>.columns.<relation_name>
+[schemas.public.views.my_view.relations.my_view_to_my_table]
+# The schema containing the referenced table or view.
+# Defaults to "public" if omitted. Must exist.
+referenced_schema = "public"
+
+# Specify either a table or view. Must exist.
+referenced_table = "my_table"
+
+# List columns in the view that reference columns in the target
```

Comment:
Have you considered a map `user_id: "id", user_name: "name"}`?

Reply:
Yes, but I don't like maps where the order matters...